source/contracts/reporting/ReportingToken.sol

```diff
@@ -63,29 +63,31 @@ contract ReportingToken is DelegationTarget, Typed, Initializable, VariableSuppl
     function redeemForkedTokens() public afterInitialized returns (bool) {
         require(market.isContainerForReportingToken(this));
         require(getUniverse().getForkingMarket() == market);
-        var _sourceReputationToken = getReputationToken();
-        var _reputationSupply = _sourceReputationToken.balanceOf(this);
-        var _attotokens = balances[msg.sender];
-        var _reporterReputationShare = _reputationSupply * _attotokens / supply;
+        IReputationToken _sourceReputationToken = getReputationToken();
+        uint256 _reputationSupply = _sourceReputationToken.balanceOf(this);
+        uint256 _attotokens = balances[msg.sender];
+        uint256 _reporterReputationShare = _reputationSupply * _attotokens / supply;
         burn(msg.sender, _attotokens);
-        var _destinationReputationToken = getUniverse().getOrCreateChildUniverse(getPayoutDistributionHash()).getReputationToken();
+        IReputationToken _destinationReputationToken = getUniverse().getOrCreateChildUniverse(getPayoutDistributionHash()).getReputationToken();
         _sourceReputationToken.migrateOut(_destinationReputationToken, this, _attotokens);
         _destinationReputationToken.transfer(msg.sender, _reporterReputationShare);
         return true;
     }
 
     // NOTE: UI should warn users about calling this before first calling `migrateLosingTokens` on all losing tokens with non-dust contents
-    // TODO: prevent calling this until all markets on the reporting window are finalized.
-    // TODO: add reporting fee distribution to this.
+    // NOTE: we aren't using the convertToAndFromCash modifier here becuase this isn't a whitelisted contract. We expect the reporting window to handle disbursment of ETH
     function redeemWinningTokens() public afterInitialized returns (bool) {
         require(market.getReportingState() == IMarket.ReportingState.FINALIZED);
         require(market.isContainerForReportingToken(this));
         require(getUniverse().getForkingMarket() != market);
         require(market.getFinalWinningReportingToken() == this);
-        var _reputationToken = getReputationToken();
-        var _reputationSupply = _reputationToken.balanceOf(this);
-        var _attotokens = balances[msg.sender];
-        var _reporterReputationShare = _reputationSupply * _attotokens / supply;
+        IReportingWindow _reportingWindow = market.getReportingWindow();
+        require(_reportingWindow.allMarketsFinalized());
+        IReputationToken _reputationToken = getReputationToken();
+        uint256 _reputationSupply = _reputationToken.balanceOf(this);
+        uint256 _attotokens = balances[msg.sender];
+        uint256 _reporterReputationShare = _reputationSupply * _attotokens / supply;
+        _reportingWindow.collectReportingFees(msg.sender, _attotokens);
```

Comment:
Security: Since this will call out to an external contract, it should be the last line of this function, after all other state updates are complete.  This should go even after the reputationToken transfer.